brave-core/src/main/java/com/github/kristofa/brave/HttpSpanCollector.java

```diff
@@ -0,0 +1,120 @@
+package com.github.kristofa.brave;
+
+import com.twitter.zipkin.gen.Span;
+import java.io.Closeable;
+import java.util.ArrayDeque;
+import java.util.LinkedList;
+import java.util.Queue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.apache.thrift.TException;
+import org.apache.thrift.protocol.TBinaryProtocol;
+import org.apache.thrift.protocol.TList;
+import org.apache.thrift.protocol.TType;
+import org.apache.thrift.transport.TMemoryBuffer;
+
+/**
+ * SpanCollector which submits spans to Zipkin, using its {@code POST /spans} endpoint.
+ */
+public class HttpSpanCollector implements SpanCollector, Closeable {
+
+  private final String baseUrl;
+  private final BlockingQueue<Span> queue;
+  private final SpanCollectorMetricsHandler metricsHandler;
+
+  private final AtomicBoolean running = new AtomicBoolean(true);
+
+  /**
+   * Create a new instance with default configuration.
+   *
+   * @param baseUrl URL of the zipkin query server instance. Like: http://localhost:9411/api/v1
+   * @param metricsHandler Gets notified when spans are accepted or dropped. If you are not
+   * interested in these events you can use {@linkplain EmptySpanCollectorMetricsHandler}
+   */
+  public HttpSpanCollector(String baseUrl, SpanCollectorMetricsHandler metricsHandler) {
+    this.baseUrl = baseUrl;
+    this.metricsHandler = metricsHandler;
+    queue = new LinkedBlockingQueue<>(1000); // if 1000 spans are pending, new spans will be dropped
+    new Thread(new SpanProcessor(), "HttpSpanCollector: " + baseUrl).start();
```

Comment:
single producer thread has pros and cons. The pros are that you can make a lot of assumptions.. and this simplifies the code design. For example, if someone wanted to have multiple collectors threads, they could make a fan-out SpanCollector that instantiates multiple of these!